### PR TITLE
Simplifies isolation mechanisms

### LINKED ISSUE #211: Inconsistent Problem Registration and Isolation Management
At the moment there are two functions to register problems: `register_isolated_function` and `register_problem`.
I suggest to rename `register_isolated_function` to `register_problem` and to remove the latter.
Further, the isolation mechanism is fairly generic: try import and isolate on failure. We should move this part to the `AbstractProblemFactory`.

### PR DESCRIPTION
**Pull request by miguelgondu**
This PR removes `register_problem` and everything associated with it.

(Closes #211 )

### CODE REVIEW

File: examples/observers/adding_an_mlflow_observer/mlflow_observer.py
Comment (RMichae1) [inline examples/observers/adding_an_mlflow_observer/mlflow_observer.py:1]: Are these MlFlow observer changes new? Why do they pop up under the problem registration refactor?

File: pyproject.toml
Comment (RMichae1) [inline pyproject.toml:1]: +1 Seems like a reasonable exclusion list.

File: examples/observers/adding_an_mlflow_observer/mlflow_observer.py
Comment (miguelgondu) [reply examples/observers/adding_an_mlflow_observer/mlflow_observer.py:1]: They were actually supposed to take place in the observer refactoring, but we forgot.

File: src/poli/objective_factory.py [hunk 1]
```diff
@@ -303,40 +164,59 @@ def create(
...
-            name,
-            seed=seed,
-            batch_size=batch_size,
-            parallelize=parallelize,
-            num_workers=num_workers,
-            evaluation_budget=evaluation_budget,
-            quiet=quiet,
-            **kwargs_for_factory,
-        )
+    # if not force_isolation:
```
Comment (RMichae1) [inline src/poli/objective_factory.py]: Is there a point in cleaning up this block? Or is this temporary?

File: src/poli/objective_factory.py [hunk 2]
```diff
@@ -303,40 +164,59 @@ def create(
...
+    #     problem = __create_problem_as_isolated_process(
+    #         name,
+    #         seed=seed,
+    #         batch_size=batch_size,
+    #         parallelize=parallelize,
+    #         num_workers=num_workers,
+    #         evaluation_budget=evaluation_budget,
+    #         quiet=quiet,
+    #         **kwargs_for_factory,
+    #     )
```
Comment (RMichae1) [inline src/poli/objective_factory.py]: End of comment block.

File: src/poli/objective_factory.py [hunk 3]
```diff
@@ -86,17 +77,31 @@ def __create_problem_from_repository(
...
-            f"Objective function '{name}' is not available in the repository."
-        )
-
+        if name in AVAILABLE_OBJECTIVES:
+            # For poli developers:
+            # If you get this error while developing a new black box,
+            # it might be because you forgot to register your problem
+            # factory and black box inside the __init__.py file of
+            # the objective_repository. Remember that you need to add
+            # them to AVAILABLE_PROBLEM_FACTORIES and AVAILABLE_BLACK_BOXES.
```
Comment (RMichae1) [inline src/poli/objective_factory.py:86]: Helpful comment:)

File: examples/the_basics/a_simple_objective_function_registration/querying_aloha.py
Comment (RMichae1) [inline examples/the_basics/a_simple_objective_function_registration/querying_aloha.py:1]: Where did this one go? Or do we not want the basic examples anymore?

File: src/poli/core/registry.py
Comment (RMichae1) [inline src/poli/core/registry.py:1]: I haven't tested the logic of these changes (yet), I'm only acknowledging that I am aware of the changes to the registry.

**Review by RMichae1 — APPROVED** [review]
See minor comments.
The updated changes are indeed better digestible.

**Pull request merged**